### ISSUE TITLE
clean construction of discrete de Rham sequences

### ISSUE THREAD

**Issue by campospinto** [pyccel/psydac#101]
it's unclear how the 'basis' argument is used in the construction of the different spaces. In different places it seems to be redundant with the kind of the spaces, in particular when using the reducing the degrees of the spaces along some axes.
The whole discretization process should be more transparent, with a clear logic.

**Comment by yguclu** [comment]
I agree. I think we can remove the `basis` argument from the `discretize_space` function, and this will reduce the confusion.

**Comment by campospinto** [comment]
We should also clarify the role of the `kind` argument in the definition of symbolic and discrete spaces, see eg in psydac/api/tests/test_api_2d_compatible_spaces.py 

Does it indicate the smoothness that is required on the functions, or also their position in some sequence ? How is the spline degree affected by this parameter ?
